internal/venice-common/src/main/java/com/linkedin/venice/ConfigKeys.java

```diff
@@ -2169,7 +2169,13 @@ private ConfigKeys() {
 
   /**
    * Controller config for the default value of {@link com.linkedin.venice.writer.VeniceWriter#maxRecordSizeBytes}.
-   * Only used in batch push jobs and partial updates.
+   * Only used in batch push jobs.
    */
   public static final String CONTROLLER_DEFAULT_MAX_RECORD_SIZE_BYTES = "controller.default.max.record.size.bytes";
+
+  /**
+   * Server config for the default value of {@link com.linkedin.venice.writer.VeniceWriter#maxRecordSizeBytes}.
+   * Only used in partial updates.
+   */
+  public static final String SERVER_DEFAULT_MAX_RECORD_SIZE_BYTES = "server.default.max.record.size.bytes";
```

Comment:
Can we merge both of these into one - `default.max.record.size.bytes`? Is there a case where both would diverge?

I suppose we also need to add some logic to handle record limit in Samza RP too as that is similar to batch push job and can chunk records on client-side ([code](https://github.com/linkedin/venice/blob/main/clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java#L397-L403))

Reply:
Yeah, it is fine to group the setting as one.
Regarding Samza RP, we don't normally put a lot of configs in Samza RP as we can't deploy it as easily as VPJ.
One thing, we can do is to compare the full value size with the limit in Venice Server for Samza RP job and fail the job if it goes beyond the limit.
If that is too complex, we can skip Samza RP job for now as there are not many RP users and they are limited into one single cluster.

Reply:
> One thing, we can do is to compare the full value size with the limit in Venice Server for Samza RP job and fail the job if it goes beyond the limit.

Yeah, I'm not sure how to do that. I do see that there is a [`StoreReponse` / `StoreInfo` from the controller in `VeniceSystemProducer`](https://github.com/linkedin/venice/blob/4a20cdf03d362989bfcdf0ee22ffd7f2038fe50a/clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java#L541)? Then to fail the job, we would just throw an `Exception`?

Reply:
I meant in `SIT`, Venice Server can throw an exception when the value size hits the limit, since the exception will be thrown before `EOP`, the Samza RP job will be killed.